client/jetpack-connect/auth-logged-in-form.jsx

```diff
@@ -284,8 +284,11 @@ class LoggedInForm extends Component {
 	}
 
 	renderNotices() {
-		const { authorizeError, queryObject, isAuthorizing, authorizeSuccess } = this.props.jetpackConnectAuthorize;
-		if ( queryObject.already_authorized && ! this.props.isFetchingSites && ! this.props.isAlreadyOnSitesList ) {
+		const { authorizeError, queryObject, isAuthorizing, authorizeSuccess, userAlreadyConnected } = this.props.jetpackConnectAuthorize;
+		if (
+			( queryObject.already_authorized && ! this.props.isFetchingSites && ! this.props.isAlreadyOnSitesList ) ||
+			( userAlreadyConnected )
```

Comment:
Yeah. I suppose I'll write some unit tests. And sorry for not writing better testing instructions.
Here is my attempt at putting this into words:

`isAlreadyOnSitesList` is not always going to work. 

In cases where a new user is trying to link to an already connected site, we will issue a request to `sites/$site` and that will come back as 200 even though the current user is not yet linked. It comes back as 200 because `sites/$site` is a public endpoint that will return data for any connected site.

Prior to this patch, upon receiving the 200, we'd add the site to the list, and then we'd disallow the new user from linking because we'd think they were already on the site.

This patch now adds a new redux action, and a new property to state. when we get the 200, we make an additional request to see if the user is actually connected. if so, we update the `userAlreadyConnected` property on the state, and disallow linkage.  if not connected, we allow the linkage, and possible remove the site from sites list.

To test `userAlreadyConnected`, you could connect a site, then try to to link an additional user to the same wpcom account.

You'll likely see the poorly worded notice :)